programs/gateway/src/instructions/update_voter_weight_record.rs

```diff
@@ -0,0 +1,58 @@
+use crate::error::GatewayError;
+use crate::state::*;
+use anchor_lang::prelude::*;
+use solana_gateway::Gateway;
+
+/// Updates VoterWeightRecord to evaluate governance power for non voting use cases: CreateProposal, CreateGovernance etc...
+/// This instruction updates VoterWeightRecord which is valid for the current Slot and the given target action only
+/// and hence the instruction has to be executed inside the same transaction as the corresponding spl-gov instruction
+#[derive(Accounts)]
+#[instruction(voter_weight_action: VoterWeightAction, target: Option<Pubkey>)]
+pub struct UpdateVoterWeightRecord<'info> {
+    /// The Gateway Registrar
+    pub registrar: Account<'info, Registrar>,
+
+    /// A gateway token from the gatekeeper network in the registrar.
+    /// Proves that the holder is permitted to take an action.
+    /// CHECK: Checked in the gateway library.
+    #[account()]
+    pub gateway_token: UncheckedAccount<'info>,
+
+    #[account(
+        mut,
+        constraint = voter_weight_record.realm == registrar.realm
+        @ GatewayError::InvalidVoterWeightRecordRealm,
+
+        constraint = voter_weight_record.governing_token_mint == registrar.governing_token_mint
+        @ GatewayError::InvalidVoterWeightRecordMint,
+    )]
+    pub voter_weight_record: Account<'info, VoterWeightRecord>,
+}
+
+pub fn update_voter_weight_record(
+    ctx: Context<UpdateVoterWeightRecord>,
+    voter_weight_action: VoterWeightAction,
+    target: Option<Pubkey>,
```

Comment:
The supplied `voter_weight` is not action/target specific and there is no need to pass them to the instruction. It should set both `weight_action` and `weight_action_target` to `None`

Reply:
Removed in the composition PR